reference/test/preconditioner/jacobi_kernels.cpp

```diff
@@ -1108,4 +1110,156 @@ TYPED_TEST(Jacobi, ConvertsEmptyToDense)
 }
 
 
+TYPED_TEST(Jacobi, ScalarL1Diag)
+{
+    /* example matrix:
+        1   1
+        2   2
+       -5       1
+        1      -1
+     */
+    using Mtx = typename TestFixture::Mtx;
+    using index_type = typename TestFixture::index_type;
+    using value_type = typename TestFixture::value_type;
+    auto mtx = Mtx::create(this->exec, gko::dim<2>{4}, 8);
+    this->template init_array<index_type>(mtx->get_row_ptrs(), {0, 2, 4, 6, 8});
+    this->template init_array<index_type>(mtx->get_col_idxs(),
+                                          {0, 1, 0, 1, 0, 2, 0, 2});
+    this->template init_array<value_type>(
+        mtx->get_values(), {1.0, 1.0, 2.0, 2.0, -5.0, 1.0, 1.0, -1.0});
+    auto diag = mtx->extract_diagonal();
+    auto diag_val = diag->get_const_values();
+
+    gko::kernels::reference::jacobi::scalar_l1(this->exec, mtx.get(),
+                                               diag.get());
+
+    EXPECT_EQ(diag_val[0], value_type{2.0});
+    EXPECT_EQ(diag_val[1], value_type{4.0});
+    EXPECT_EQ(diag_val[2], value_type{6.0});
+    EXPECT_EQ(diag_val[3], value_type{2.0});
+}
+
+
+TYPED_TEST(Jacobi, BlockL1)
+{
+    /* example matrix:
+        1   1
+        2   2
+       -5       1
+        1      -1   1
+     */
+    using Mtx = typename TestFixture::Mtx;
+    using index_type = typename TestFixture::index_type;
+    using value_type = typename TestFixture::value_type;
+    auto mtx = Mtx::create(this->exec, gko::dim<2>{4}, 9);
+    this->template init_array<index_type>(mtx->get_row_ptrs(), {0, 2, 4, 6, 9});
+    this->template init_array<index_type>(mtx->get_col_idxs(),
+                                          {0, 1, 0, 1, 0, 2, 0, 2, 3});
+    this->template init_array<value_type>(
+        mtx->get_values(), {1.0, 1.0, 2.0, 2.0, -5.0, 1.0, 1.0, -1.0, 1});
+    gko::array<index_type> block_ptr(this->exec, {0, 1, 2, 4});
+
+    gko::kernels::reference::jacobi::block_l1(this->exec, 3, block_ptr,
+                                              mtx.get());
+
+    GKO_ASSERT_MTX_NEAR(mtx,
+                        l({{2.0, 1.0, 0.0, 0.0},
+                           {2.0, 4.0, 0.0, 0.0},
+                           {-5.0, 0.0, 6.0, 0.0},
+                           {1.0, 0.0, -1.0, 2.0}}),
```

Comment:
This doesn't seem right. Are the off-diagonal values of the diagonal block also added to the diagonal of the diagonal block? IMO that shouldn't be the case, only values from the off-diagonal blocks should be used (at least according to the original reference).

Reply:
I think it is the case, right?
the blocks here are (0, 0), (1, 1), and (2:4, 2:4)

Reply:
You're right, I didn't understand the blocking of the matrix. Maybe the blocks can be denoted in the comment above?